x/ccv/provider/keeper/relay.go

```diff
@@ -160,6 +169,46 @@ func (k Keeper) EndBlockVSU(ctx sdk.Context) {
 		// the updates will remain queued until the channel is established
 		k.SendVSCPackets(ctx)
 	}
+
+	return valUpdates, nil
+}
+
+// ProviderValidatorUpdates returns changes in the provider consensus validator set
+// from the last block to the current one.
+// It retrieves the bonded validators from the staking module and creates a `ConsumerValidator` object for each validator.
+// The maximum number of validators is determined by the `maxValidators` parameter.
+// The function returns the difference between the current validator set and the next validator set as a list of `abci.ValidatorUpdate` objects.
+func (k Keeper) ProviderValidatorUpdates(ctx sdk.Context) []abci.ValidatorUpdate {
```

Comment:
Is this function tested?

Reply:
It's hit in tests, yes, in particular many integration tests pass through this.
Do you think a separate unit test is necessary? I think it doesn't provide a ton of value, since DiffValidators is tested, and this function doesn't really have any logic other than that call,
so I'm happy with how it's tested, but I can add a test just for this if your call is that it's important to have it.

Reply:
Added a unit test for the function anyways :)